torch_xla/csrc/reduction.cpp

```diff
@@ -73,7 +72,7 @@ xla::XlaOp CreateSummation(
   xla::XlaOp result = xla::Reduce(
       input, init_value, XlaHelpers::CreateAddComputation(shape.element_type()),
       dimensions);
-  if (scale && rinfo.element_count > 1) {
```

Comment:
Are you sure?
Coming in with count==0 means division by zero below...

Reply:
yea `nan` is the expected input for this case: in the sense that numpy and pytorch both returns nan...
```
In [1]: import numpy as np

In [2]: import torch

In [3]: a = torch.rand(2, 0, 4)

In [4]: a.mean()
Out[4]: tensor(nan)

In [5]: b = np.random.rand(2, 0, 4)

In [6]: b.mean()
/home/ubuntu/miniconda3/envs/maskrcnn36/bin/ipython:1: RuntimeWarning: Mean of empty slice.
  #!/home/ubuntu/miniconda3/envs/maskrcnn36/bin/python
/home/ubuntu/miniconda3/envs/maskrcnn36/lib/python3.6/site-packages/numpy/core/_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
Out[6]: nan
```

Reply:
What about the line:
```Python
1.0f / static_cast<float>(rinfo.element_count)
```

?

Reply:
sorry I didn't get this - what's the issue about that line? It produces a nan which is then multiplied to result.

Reply:
Division by zero is Undefined Behavior, and many times depends on compiler options (whether to SIGFPE or not).
So better not rely on it returning NaN.

Reply:
ehhh what's a better way to get nan in this case?

Reply:
Not sure what the most reliable way of construction NaN in XLA. I assume `ConstantR0` with a NaN float value should work.

Reply:
ScalarValue already does that.
I am not sure how XLA "feels" about NaN uploaded to TPU device. We can check.
In any case, that should be:
```C++
rinfo.element_count > 0 ? 1.0f / static_cast<float>(rinfo.element_count) : NAN
```